libcontainer/cgroups/v1_utils.go

```diff
@@ -99,11 +101,29 @@ func tryDefaultPath(cgroupPath, subsystem string) string {
 // expensive), so it is assumed that cgroup mounts are not being changed.
 func readCgroupMountinfo() ([]*mountinfo.Info, error) {
 	readMountinfoOnce.Do(func() {
-		cgroupMountinfo, readMountinfoErr = mountinfo.GetMounts(
+		// mountinfo.GetMounts uses /proc/self/mountinfo, however in runc we
+		// have threads that join different namespaces and thus the output of
+		// the thread-group's mountinfo can depend on Go scheduling. Instead,
+		// use thread-self as we know the current thread has the "correct"
+		// mount namespace.
+		//
+		// See <https://github.com/moby/sys/pull/128>.
```

Comment:
It's a shame that PR is still not merged. Let's merge it and rely on updated mountinfo instead of adding this here.

Reply:
@cyphar please vendor https://github.com/moby/sys/releases/tag/mountinfo%2Fv0.7.0

Reply:
It seems that the `/proc/self/mountinfo` fallback (that we have in runc) is needed for our checkpoint tests -- #4109 is failing for this reason. I've opened moby/sys#135 to fix it.

Reply:
Also, a `moby/sys` update is not a blocker for this PR -- I just mentioned it because with the right updates it wouldn't be necessary to use `ProcThreadSelf` here.